packager/media/chunking/text_chunker.cc

```diff
@@ -92,7 +92,11 @@ Status TextChunker::OnTextSample(std::shared_ptr<const TextSample> sample) {
     RETURN_IF_ERROR(DispatchSegment(segment_duration_));
   }
 
-  samples_in_current_segment_.push_back(std::move(sample));
+  if (sample->duration() > 0) {
+    samples_in_current_segment_.push_back(std::move(sample));
+  } else {
+    sample.reset();  // duration == 0 means heartbeat. drop it.
+  }
```

Comment:
@marcusspangenberg Is reset the right thing to do?

Reply:
Since you have a reference to sample that is local to this function scope (you receive a copy of a shared_ptr as argument) the sample.reset() statement only releases your local reference. That reference will be released when you return from the function regardless so the reset() statement is redundant and probably does not do what you wanted.